docs/rholang/rholangtut-0.3_ES.md

```diff
@@ -0,0 +1,402 @@
+
+5) Antes de esta línea, no hay mensajes en el canal `valueStore`. Después de enviar el valor inicial, este es el único valor en ese canal.
+
+6) Creamos un contrato para escuchar en el canal `get`. Cada vez que se envía un mensaje en `get`, el cuerpo del contrato se ejecutará
+
+7) Bloqueamos hasta que recibimos un mensaje del canal `valueStore`. Debido a que hay como mucho un mensaje siempre esperando en `valueStore`, leer el mensaje se comporta de forma muy similar a la obtención de un candado (cierre de exclusión mutua).
+
+8) Enviamos nuevamente el valor actual en `valueStore`, permitiendo que se procesen otros mensajes (es decir, retirando el cierre), y enviamos el valor actual de vuelta al cliente en el canal `ack`.
+
+11) Al mismo tiempo que el contrato `get`, ejecutamos un contrato escuchando en `set`.
+
+12) Bloqueamos hasta que aparezca un mensaje en `valueStore`, y entonces lo leemos. Tiramos el mensaje que leemos.
+
+13) Enviamos el nuevo valor para guardarlo en `valueStore` e indicamos que la operación está completa.
+
+18-36) El código de uso demuestra la creación de una celda, la asignación del valor inicial 123, la obtención de ese valor, el establecimiento de ese valor en 456, y después la obtención de ese valor.
+
+Destacan las capas profundas de devolución de llamada (callback). Rholang fue diseñado para hacer que los cálculos concurrentes fueran expresados de forma natural. Como consecuencia, las dependencias de datos implícitas en la secuenciación en otros lenguajes deben hacerse explícitas.
+
+## Iteración y coincidencia
+
+En el siguiente código, mostramos un ejemplo de iteración mediante una lista.
+
+     1	new iterate in {
+     2	  contract iterate(@list, process, done) = {
+     3	    match list {
+     4	      [hd, ...tl] => {
+     5	        new ack in {
+     6	          process!(hd, *ack) |
+     7	          for (_ <- ack) { iterate!(tl, *process, *done) }
+     8	        }
+     9	      }
+    10	      _ => done!(Nil)
+    11	    }
+    12	  } |
+    13	  new process, done in {
+    14	    iterate!([4,5,6], *process, *done) |
+    15	    contract process(@item, ack) = {
+    16	      /* manejo del procesamiento del elemento */
+    17	      ack!(Nil)
+    18	    } |
+    19	    for (_ <- done) {
+    20	      /* listo! */
+    21	      Nil
+    22	    }
+    23	  }
+    24	}
+
+3) La construcción `match` permite desestructurar una variable mediante la coincidencia de patrones.
+
+4) Los patrones de la lista facilitan la coincidencia con el resto de una lista. Si `list` coincide con el patrón de un par head/tail ejecutamos el cuerpo principal del ciclo.
+
+5) Creamos un canal para que el handler de elementos nos notifique que ya está hecho con el elemento actual.
+
+6) Llamamos al procesador en el elemento y el canal de acuse de recibo.
+
+7) Cuando recibimos el acuse de recibo, volvemos a llamar al iterador a la cola.
```

Comment:
You sometimes use "head/tail" and sometimes "cola" etc.  Please be consistent in translated vs. literal terms.

Reply:
Fixed as well!